pkg/controller/remotemachineset/awsactuator.go

```diff
@@ -349,3 +380,18 @@ func (a *AWSActuator) getSubnetsByAvailabilityZone(pool *hivev1.MachinePool) (ma
 
 	return subnetsByAvailabilityZone, nil
 }
+
+func isUsingUnsupportedSpotMarketOptions(pool *hivev1.MachinePool, clusterVersion string, logger log.FieldLogger) bool {
+	if pool.Spec.Platform.AWS.SpotMarketOptions == nil {
+		return false
+	}
+	finalizedVersion, err := semver.FinalizeVersion(clusterVersion)
+	if err != nil {
+		logger.WithError(err).WithField("clusterVersion", clusterVersion).Warn("could not finalize the cluster version")
+	}
```

Comment:
why not return after logging the error here?

Reply:
It should return. I messed it up during a couple different approaches that I took here. Good catch.

Reply:
I added a unit test for this. In doing so, I realized that `FinalizeVersion` does a `Parse`, meaning that the subsequent `ParseTolerant` will never return an error. I changed the code so that it does not use `FinalizeVersion` and instead directly removes the parts other than major, minor, and patch from the parsed version.